pkg/account/deployer/client.go

```diff
@@ -247,6 +249,168 @@ func (d *accountManagementClient) upsertUser(ctx context.Context, userId string)
 	return userId, nil
 }
 
+func (d *accountManagementClient) upsertServiceUser(ctx context.Context, serviceUserId string, data ServiceUser) (remoteId, error) {
+	if serviceUserId == "" {
+		suId, err := d.getServiceUserIDByName(ctx, data.Name)
+		if err != nil {
+			var rnfErr *ResourceNotFoundError
+			if !errors.As(err, &rnfErr) {
+				return "", err
+			}
+
+			return d.createServiceUser(ctx, data)
+		}
+		serviceUserId = suId
+	}
+
+	return d.updateServiceUser(ctx, serviceUserId, data)
+}
+
+func (d *accountManagementClient) createServiceUser(ctx context.Context, dto accountmanagement.ServiceUserDto) (string, error) {
+	uuidDto, resp, err := d.client.ServiceUserManagementAPI.CreateServiceUserForAccount(ctx, d.accountInfo.AccountUUID).ServiceUserDto(dto).Execute()
+	defer closeResponseBody(resp)
+	if err = handleClientResponseError(resp, err, "failed to create service user"); err != nil {
+		return "", err
+	}
+
+	if uuidDto == nil {
+		return "", errors.New("the received data are empty")
+	}
+
+	return uuidDto.Uuid, nil
+}
+
+func (d *accountManagementClient) updateServiceUser(ctx context.Context, serviceUserId string, dto accountmanagement.ServiceUserDto) (string, error) {
+	resp, err := d.client.ServiceUserManagementAPI.UpdateServiceUserForAccount(ctx, d.accountInfo.AccountUUID, serviceUserId).ServiceUserDto(dto).Execute()
+	defer closeResponseBody(resp)
+
+	// handle a 404 here if need be as handleClientResponseError discards it!
+	if is404(resp) {
+		return "", ResourceNotFoundError{Identifier: serviceUserId}
+	}
+
+	if err = handleClientResponseError(resp, err, "failed to update service user"); err != nil {
+		return "", err
+	}
+
+	return serviceUserId, nil
+}
+
+// ResourceNotFoundError is an error signifying that the desired resource was not found.
+type ResourceNotFoundError struct {
+	Identifier string
+}
+
+func (e ResourceNotFoundError) Error() string {
+	return fmt.Sprintf("resource '%s' not found", e.Identifier)
+}
+
+func (c *accountManagementClient) getServiceUserIDByName(ctx context.Context, name string) (string, error) {
```

Comment:
I see that once you're using it with d and once with c, if you could rename that to all be the same.

Reply:
I'll go with `c` for the moment, one step close towards having a shared client 👍

Reply:
Or how about `this`? 😅